src/meta/meta_ckpt.c

```diff
@@ -963,11 +1143,32 @@ __wt_meta_ckptlist_free(WT_SESSION_IMPL *session, WT_CKPT **ckptbasep)
     if ((ckptbase = *ckptbasep) == NULL)
         return;
 
-    WT_CKPT_FOREACH (ckptbase, ckpt)
+    /*
+     * Sometimes the checkpoint list has a checkpoint which has not been named yet, but carries an
+     * order number.
+     */
+    WT_CKPT_FOREACH_NAME_OR_ORDER (ckptbase, ckpt)
         __wt_meta_checkpoint_free(session, ckpt);
     __wt_free(session, *ckptbasep);
 }
 
+/*
+ * __wt_meta_saved_ckptlist_free --
+ *     Discard the saved checkpoint list.
+ */
+void
+__wt_meta_saved_ckptlist_free(WT_SESSION_IMPL *session)
+{
+    WT_BTREE *btree;
+
+    btree = S2BT_SAFE(session);
+    if (btree == NULL)
+        return;
+
```

Comment:
Why are we using `S2BT_SAFE` here? Is there a caller that might require that check? It seems like calling here without a valid WT_BTREE handle would indicate a bug?

If this is correct, I think it deserves a comment as to why.

Reply:
Assuming the assert just drops core, you may not need one, given indirection through the NULL pointer will do the same thing?

Reply:
correct, removing assert.